src/data_storage/data_source.py

```diff
@@ -15,3 +22,26 @@ def get_node(self, story_id: str, uri: str) -> Node:
     @abc.abstractmethod
     def get_story_list(self) -> StoryList:
         """Get list of stories"""
+
+    def create_story(self, data: dict):
+        """
+        Template method.
+        Raise exception if unsuccessful.
+        Validate and create story item
+        """
+        validated_story_data = parse_story_structure(data)
+        graph, root_node = parse_graph(validated_story_data)
+        self._validate_story_item(graph, root_node)
+        self._create_story(validated_story_data)
+        return validated_story_data
+
+    @abc.abstractmethod
+    def _create_story(self, data):
+        raise NotImplementedError
```

Comment:
For `abstractmethod` we don't have to raise `NotImplementedError`. Instead we can add docstring here.